zerver/lib/rest.py

```diff
@@ -40,10 +41,27 @@ def rest_dispatch(request, **kwargs):
     etc, as that is where we route HTTP verbs to target functions.
     """
     supported_methods = {} # type: Dict[str, Any]
+
+    # Flags can be attached to specific methods by passing a tuple instead of a string to the
+    # methods map in rest_dispatch based url pattern.
+    #
+    # The tuple will be unpacked as a str,list with the string representing the view finction
+    # and the list representing the list of flags.
+    flags = {} # type: Dict[str, List[Any]]
+
     # duplicate kwargs so we can mutate the original as we go
     for arg in list(kwargs):
         if arg in METHODS:
-            supported_methods[arg] = kwargs[arg]
+            entry = kwargs[arg]
+            if isinstance(entry, tuple):
+                target_function, view_flags = entry
+                supported_methods[arg] = target_function
+                for flag in view_flags:
+                    if flag in FLAGS:
+                        flags[arg] = flag
+            else:
+                target_function = entry
+                supported_methods[arg] = target_function
```

Comment:
Looks like you missed my comment suggestion you move this logic the `if method_to_use` statement to simplify it?